Made it compatible with other devices

SylvainGa opened the pull request.

SylvainGa committed:
Made it compatible with other devices

SylvainGa committed:
Delete fonts.xml

SylvainGa committed:
Delete .gitignore

SylvainGa committed:
Create .gitignore

SylvainGa committed:
Added a Tesla battery level as an indicator

SylvainGa committed:
Better handling of errors and can now refresh an access token through the help of the refresh token (hopefully)

SylvainGa committed:
Will show by a '+' sign after the % when the car is charginh

SylvainGa committed:
Better handling of 408 error

SylvainGa committed:
Not as chatty debugging code and needs to handle more than one event per call of onTemporalEvent

SylvainGa committed:
More debugging code

SylvainGa committed:
Now display a 's' after the % if the car is sleeping

SylvainGa committed:
Now  preempt token expiry and ask for a new one if it's about to expire

SylvainGa committed:
Fixed bug 210

SylvainGa committed:
More elegant and faster way of testing if the token is empty

SylvainGa committed:
Started getting -403 error so ended up removing some debugging code to free space

SylvainGa committed:
Moved from vehicle_state to charge_state which receives 1.5KB of data instead of 5KB.

Ran out of background process space and this data, converted to a dictionary was even taking more space and was too big for the space left in the background process.

SylvainGa committed:
Revert "Moved from vehicle_state to charge_state which receives 1.5KB of data instead of 5KB."

This reverts commit 47750515f52edfd0a54c6a773162ddba2ff1b6be.

SylvainGa committed:
Revert "Started getting -403 error so ended up removing some debugging code to free space"

This reverts commit 6524842a9ebf498bfe38c51eed32f636d0537bbb.

SylvainGa committed:
Revert "More elegant and faster way of testing if the token is empty"

This reverts commit 7d6cc11a404a8abb7459966e4baf301b011f883a.

SylvainGa committed:
Revert "Fixed bug 210"

This reverts commit 1c77cf82485307161e45a960682a3358419a166f.

SylvainGa committed:
Revert "Now  preempt token expiry and ask for a new one if it's about to expire"

This reverts commit ade7252f5f0880070e789df3f809a1fa451e66e2.

SylvainGa committed:
Removed most of the stuff I did yesterday because I hit a memory limitation (I think). Stable now. Log verbosity decreased. Can almost keep an hour worth of output  on the watch .TXT files themselves

SylvainGa committed:
Debug code to show when the access token expires

SylvainGa committed:
Reinstate the preemptive refresh close to the expiration time. If we launch while the car is asleep, use its last known battery charge

SylvainGa committed:
Removed some debugging line and receiving a token no longer stales the data

SylvainGa committed:
Now toggles between battery charge, sentry mode, preconditioning and inside temperature

SylvainGa committed:
Adjustment to the display times of the Tesla stuff

SylvainGa committed:
Rewritten Propertie TeslaBatterieLevel to TeslaInfo since it does more than just the battery stuff now

SylvainGa committed:
Crash in debug code when starting from fresh

SylvainGa committed:
Fix code to display if we have an error, fix code that disables TeslaInfo when not selected, fix code that always  display P and S using theme color. Added more watches by using a more recent SDK version.

SylvainGa committed:
Removal of debug code to reduce the size of the executable and additions of new devices

SylvainGa committed:
Addition of Epix2 to the supported watches

SylvainGa committed:
Disables background service on watches that doesn't support it.

SylvainGa committed:
Fix sunrise showing todays instead of tomorrow when sunset has been reached

SylvainGa committed:
GarminWeather can be used instead of OpenWeatherMap

SylvainGa committed:
Tidying the code a bit

SylvainGa committed:
Modification to the Garmin Weather code so it can run on most 64KB watches

SylvainGa committed:
Fixed a crash when no Garmin Weather data returned. Associated the global GPS var to what was returned by Garmin Weather, removed old debig code and added comments specifying where to cut to remove Tesla addition

Cut between
//****************************************************************
//******** REMVOVED THIS SECTION IF TESLA CODE NOT WANTED ********
//****************************************************************
and
//****************************************************************
//******************** END OF REMVOVED SECTION *******************
//****************************************************************

SylvainGa committed:
Update DataFields.mc

SylvainGa committed:
Update to altitude code to keep previous value if GPS is lost

SylvainGa committed:
Removal of Garmin Swim 2

SylvainGa committed:
Merge branch 'master' of https://github.com/SylvainGa/crystal-face

SylvainGa committed:
Second attempt at fixing the Altitude code so it stays put, even if the watch face loses focus. Also fix a very rare occurrence of a crash when displaying seconds and the area where the seconds are to be displayed is overwritten by the move bar

SylvainGa committed:
Fixed the seconds not displaying by code introduced to fix a crash on a simulated FR55 when in low power mode and seconds should be displayed

The real cause of the error is in the 208x208 resource file here

		<!-- Partial updates not supported -->
		<!--param name="secondsClipY">139</param-->
		<!--param name="secondsClipWidth">22</param-->
		<!--param name="secondsClipHeight">15</param-->

With the variables commented out and the FR55 supporting 'partial update', it crashes. No one complained about the frozen seconds in low power mode so I'm leaving it like that for the time being. Might fix later.

SylvainGa committed:
Add floors climbed as a data field

SylvainGa committed:
Fix the '/' in the titillium-web-semibold-28 font in 280x280 plus added the '/' to the NormalFontCities

SylvainGa committed:
Added Descent Mk1, Mk2 and MK2S as well as 'Do Not Disturb' as indicator

SylvainGa committed:
AOD Shows Tesla's battery level

SylvainGa committed:
Added images for ConnectIQ

SylvainGa committed:
Removal of watches  with not enough memory to run reliably

SylvainGa committed:
Added SolarIntensity  as a data field

SylvainGa committed:
Added Body Battery as a data field. Bug fix for Solar Intensity on 390x390 watches

SylvainGa committed:
Addition of Forerunner 255, 255 Music, 255s, 255s Music

SylvainGa committed:
Text that needs to be translated

SylvainGa committed:
Fixed 'Û' missing in the 280x280 font file.

ChristopheMB committed:
Fix Dutch translations

Correct the spelling of some Dutch translations, add missing translations.

ChristopheMB committed:
Merge pull request #1 from ChristopheMB/ChristopheMB-patch-1

Fix Dutch translations

SylvainGa committed:
Added the Enduro, Venu SQ 2 and Venu SQ 2 Music

SylvainGa committed:
Merge pull request #6 from ChristopheMB/master

Fix Dutch translations

SylvainGa committed:
Fixed AlwaysOnDisplay of Venu SQ 2 and Venu SQ 2 Music

SylvainGa committed:
Added Recovery Time left as a data field

SylvainGa committed:
Fixed 30 minutes offset time zone, allows Tesla's server to be specified to allow Mainland China and added customizable colors

SylvainGa committed:
Update README.md

Too many changes to show in the Garmin "what's new" section so I moved them here instead.

SylvainGa committed:
Updated App version

SylvainGa committed:
Merge branch 'master' of https://github.com/SylvainGa/crystal-face

SylvainGa committed:
Update README.md

SylvainGa committed:
Update README.md

SylvainGa committed:
Update README.md

SylvainGa committed:
Update README.md

SylvainGa committed:
Update README.md

SylvainGa committed:
Update README.md

SylvainGa committed:
Removed troubleshooting code left by mistake

SylvainGa committed:
Merge branch 'master' of https://github.com/SylvainGa/crystal-face

SylvainGa committed:
Update README.md

SylvainGa committed:
Added BodyBattery as a Goal Meter

SylvainGa committed:
Added Body Battery as a Goal Meter

SylvainGa committed:
Remove warning at build time

SylvainGa committed:
"Hide seconds" has been changed from a checkbox to a list with three entries

  - Show seconds
  - Hide seconds when inactive
  - Always hide seconds

Show and Hide seconds behaves like the old toggle but "Hide seconds when inactive" will not display seconds when the watchface is in low power mode (ie, inactive). "Memory-In-Pixel" watch consumes energy when a pixel is changed. Preventing the seconds to update every secondes can help reduce the energy drawn by the watchface. Has no Effect on AMOLED watches.

SylvainGa committed:
Update README.md

SylvainGa committed:
Update README.md

SylvainGa committed:
Update README.md

SylvainGa committed:
Compiled with Connect IQ 4.2.1 which added the Forerunner 265, 265s and 965 and removal of dependencies for devices

SylvainGa committed:
Revert "Compiled with Connect IQ 4.2.1 which added the Forerunner 265, 265s and 965 and removal of dependencies for devices"

This reverts commit 1f84ed660bbc095facf77d1d605bf57a391c4f61.

SylvainGa committed:
Revert "Revert "Compiled with Connect IQ 4.2.1 which added the Forerunner 265, 265s and 965 and removal of dependencies for devices""

This reverts commit 4c00ee32c168791d14a9052097156905b1ace16e.

SylvainGa committed:
X

SylvainGa committed:
Revert "Compiled with Connect IQ 4.2.1 which added the Forerunner 265, 265s and 965 and removal of dependencies for devices"

This reverts commit 1f84ed660bbc095facf77d1d605bf57a391c4f61.

SylvainGa committed:
Revert "X"

This reverts commit 60dd0cec3b59187c08c77a396165b453b3f7d15d.

SylvainGa committed:
Brought back device dependencies. Added BodyBattery to FR55 and adjusted the 454x454 objects location on screen

SylvainGa committed:
Update README.md

SylvainGa committed:
Removed two unused variables

SylvainGa committed:
Update monkey.jungle

SylvainGa committed:
2.14.2

- Added Longitude and Latitude as editable fields in the parameters to allow for a DEFAULT location for sunrise/sunset and weather. Any activity found with GPS data will override this location. This should fix many "gps?" error. Keep in mind it can take up to 5 minutes before data is retrieved from OWM or Garmin Weather.
- Added BodyBattery as a datafield to the Forerunner 55

SylvainGa committed:
Update README.md

SylvainGa committed:
Increased time for for 454x454 watch and check for phone connection to gray out  weather icon

SylvainGa committed:
Update README.md

SylvainGa committed:
Increased 454x454 devices time and date font

SylvainGa committed:
Update README.md

SylvainGa committed:
Update README.md

SylvainGa committed:
2.15.0

- Added Stress Level as a goal meter and datafield
- Consolidated some of the watches that were missing goal meters or datafields. If you select one and get a gray icon, it means it's not available for your watch model. If there is a goal meter or datafield that your watch support but can't see it, let me know.

SylvainGa committed:
Update README.md

SylvainGa committed:
Update README.md

SylvainGa committed:
2.15.1

- Reverted to Connect IQ 4.1.7 since there are too many problem with sensors with this 4.2.x version so far. Hopefully this doesn't break the newest Forerunner 265, 265S and 965.
- Moved the line between time and date in Always so it doesn't strike the time.

SylvainGa committed:
Update README.md

SylvainGa committed:
2.15.2

- Access to stress level is not as stable as the Garmin access to it. To help here, if a value isn't returned, the last good one is used and the icon is greyed out to indicate the data is stale.
- Bug fix in the default latitude/longitude when the last activity has no GPS coordinates.

SylvainGa committed:
2.16.0

- Added a optional colon between hours and minutes

SylvainGa committed:
Fixed the Venu2 SQ font

SylvainGa committed:
Reformat comments and reactivate current debug one

SylvainGa committed:
Update CrystalView.mc

SylvainGa committed:
Cleaned up strings.xml, reorganized properties.xml to match order in settings.xml. Version number taken from resource\strings\strings.xml

SylvainGa committed:
Update .gitignore

SylvainGa committed:
Fixed the missing "+" character in the semibold 28pt font.

SylvainGa committed:
Merge branch 'master' of https://github.com/SylvainGa/crystal-face

SylvainGa committed:
Removed manifest.xml from .gitignore

SylvainGa committed:
Need to think to update the version before compiling

SylvainGa committed:
Update README.md

SylvainGa committed:
No point building betas for watches I don't have

SylvainGa committed:
Changed setProperties/getProperties for setValue/getValue and limited lat an lng to 5 decimal when storing in Properties

 I saw a Lat of 45.3923803809328 which prevented the settings from saving.

SylvainGa committed:
Update DataFields.mc

SylvainGa committed:
2.16.5 - Fixed a crash when entering a City in the "Add a local time in City"

SylvainGa committed:
2.16.5a - Fixed two crashes reported through the Error Reporting Application

SylvainGa committed:
2.16.5b Fixed a crash reported through the Error Reporting Application (ERA)

SylvainGa committed:
Update strings.xml

SylvainGa committed:
V2.17.0 interim

- Tesla-Info now support launching the main Tesla-Link App from the watch face.
- Heart rate, PulseOx, BodyBattery and Stress Level are now read through complications if available
- Complication read fields will launch their widget when long pressed. Applies to Fields, Indicators (for Tesla-Info only for now) and Goals.

SylvainGa committed:
V2.17.0 released

SylvainGa committed:
Reinstated logging after release

SylvainGa committed:
Prepping for V2.17.1

SylvainGa committed:
Fixed step and floor goals for Complication since they don't have any (still has to rely on ActivityMonitor for that)

SylvainGa committed:
One more crash because mDrawable is null in AoD

SylvainGa committed:
Made Complications optional, added some sanity check around Goals values to prevent crashes

SylvainGa committed:
V2.17.2

- Made Complications an optional item, disabled by default since some watches are not reading them properly. Settable through Settings
- Adjusted the touch points around the Fields, Indicators and Goals value.
- Increased the size of the touchpoints as the number of Fields on screen decreases (ie, smallest touchpoint boxes sizes are when three Fields are displayed side by side and the largest touchpoint box is when only one field is dislayed). Doesn't apply to Indicators, just Fields.
- Fixed a few crashes reported through the Error Reporting Application (ERA)

SylvainGa committed:
Trying to fix wierd AoD crash when main display activated

Still a work in progress, I can't reproduce it on my Venu 2

SylvainGa committed:
More fix before releasing V2.17.2

Move Complications from mDrawables to mGoalTypes, similar to mFieldTypes

SylvainGa committed:
Fixed a crash when Theme color override was invalid

SylvainGa committed:
V2.17.2

- Made Complications an optional item, disabled by default since some watches are not reading them properly. Settable through Settings
- Adjusted the touch points around the Fields, Indicators and Goals value.
- Increased the size of the touchpoints as the number of Fields on screen decreases (ie, smallest touchpoint boxes sizes are when three Fields are displayed side by side and the largest touchpoint box is when only one field is dislayed). Doesn't apply to Indicators, just Fields.
- Fixed a few crashes reported through the Error Reporting Application (ERA) including a crash when Theme color override was invalid. If the color stays to what the selected Theme color is, it's because your override is invalid

SylvainGa committed:
Replaced the 'T' symbol for Tesla in the AoD for the real Tesla icon

SylvainGa committed:
Hardened the Properties.getValue calls

SylvainGa committed:
2.17.4 - One more attempt at fixing the crash because of the Settings change since 2.17.1

SylvainGa committed:
Changed version in string file

SylvainGa committed:
Optimize global function calling time, fix bug in themelight dectection

SylvainGa committed:
New Global.mc file. Moved stuff away from App class. Starting new Tesla code

SylvainGa committed:
Redid Tesla code. Fixed Steps Complication where above 10,000 becomes 10.000.

SylvainGa committed:
Added the TeslaLink property

SylvainGa committed:
Do not read the whole weather map from storage every seconds, only when we got new data

SylvainGa committed:
Updated the version to 2.17.5

SylvainGa committed:
Code finished, hopefully

SylvainGa committed:
Starting implementing weather Complications

SylvainGa committed:
Weather, sunrise, recovery time addition and bug fix in weather

Added Weather and Sunrise as touchpoints.
Added RecoveryTime as Complication and touch point.Fixed a crash in weather when temperature is null

SylvainGa committed:
Moved Weather calculation to onExitSleep for efficiency and fixed bug in updateComplications

SylvainGa committed:
RecoveryTime from complication is in minutes, not second, and Tesla handling for 'offline'

SylvainGa committed:
Bug fix in TeslaVehicleIndex and added that property

SylvainGa committed:
Update README.md

SylvainGa committed:
New code for offline vehicle

SylvainGa committed:
Tweaking what shows up when an error is received

SylvainGa committed:
Done

SylvainGa committed:
Showing when Tesla Complication received

SylvainGa committed:
Had to reinstate the original sunrise/sunset routine because some watches are still on 3.2.0

SylvainGa committed:
Consolidated debug statements

SylvainGa committed:
Some debug fix

SylvainGa committed:
Update manifest.xml

SylvainGa committed:
V2.18.1

- Gave the weather station name its own line so it doesn't hide the Goals current/max fields. Local City Time will still hide these though. This is how it always was.
- Added an option in Settings (Show weather station name) to show or not the weather station name, defaults to True.
- Watches that although are at CIQ V3.2.0 and above but are still lacking Garmin Weather will display "N/A" for temperature/humidity/sunrise/sunset.
- Trying to free some more room so watches with 98KB of watch face memory doesn't crash with Out of Memory errors.

SylvainGa committed:
Gave the Forerunner 55 access to the weather and humidity since it supports Garmin Weather

SylvainGa committed:
Needed for the 240 x 240 watches

SylvainGa committed:
V2.18.2

- Fixed a crash on a VenuSQ
- Fixed a crash when invalid data is received from the background process

SylvainGa committed:
Reinstated debug statements

SylvainGa committed:
Reinstated debug statements

SylvainGa committed:
For 2.18.3

- Corrected invalid value for Stress read from sensor, not Complications
- For performance, ignore Tesla-Link Complications if we're told not to use them
- Changed the logic for when the background process should run next

SylvainGa committed:
Fixed AOD and Tesla battery not showing and added debug log message for Temporal Event time triggering

SylvainGa committed:
Made some annotation to help devices with low memory to exclude stuff they can't use (Complications for this round)

SylvainGa committed:
Saw a weird crash in onPress so I added type checking to prevent it from happening again

SylvainGa committed:
Reduced the logMessage quantity to get more lines in  log file

SylvainGa committed:
Fix for mDigitStyle and mShowWeatherStationName not included when in :noComplications

SylvainGa committed:
try/catch to prevent out of memory in Bg.exit

SylvainGa committed:
2.18.6

- Added a default custom Complication to launch when pressing anywhere that's it's not a field, indicator or goal configured as a complication. By Default, this field is set to "Flashlight" (new Widget I created), which if installed on the watch, will transform the screen into a flashlight with three intensity, with the launched one being the dimmest
- Fixed a crash in onPress

SylvainGa committed:
Update BackgroundService.mc

SylvainGa committed:
Added support for new watches

SylvainGa committed:
Added support for the Venu3 and Venu3S

AdrienChenu reviewed (commented):
Longitudes go from -180 to +180. This current check restrict the time zone east of Chicago and west of India (roughly).

The local city feature does not currently work for East Asia (China, Japan) or California for example.

Hopefully this change will fix this. 
Thanks for considering it.

AdrienChenu reviewed (commented):
Longitudes go from -180 to +180. This current check restrict the time zone east of Chicago and west of India (roughly).

The local city feature does not currently work for East Asia (China, Japan) or California for example.

Hopefully this change will fix this. 
Thanks for considering it.

SylvainGa committed:
2.18.8 - Added support for the VivoActive5

SylvainGa commented:
Hi, I haven't had the time to look at it yet but should be able before the
end of the week.

Le lun. 18 sept. 2023, à 17 h 46, Adrien Chenu ***@***.***> a
écrit :

> ***@***.**** commented on this pull request.
>
> Longitudes go from -180 to +180. This current check restrict the time zone
> east of Chicago and west of India (roughly).
>
> The local city feature does not currently work for East Asia (China,
> Japan) or California for example.
>
> Hopefully this change will fix this.
> Thanks for considering it.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/warmsound/crystal-face/pull/209#pullrequestreview-1632029382>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ALYXESXLNTMTB5ETZDWHHT3X3C6LLANCNFSM5ESSMNUQ>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

AdrienChenu committed:
Extend Longitude to between -180 to +180

AdrienChenu commented:
Thank you. No rush at all.

I'm quite new to contributing to Github, I thought my comment was linked to the line of code when I wrote it, and I posted the same comment twice. Sorry. 😅

I managed to create a Pull Request that should hopefully fix it: https://github.com/SylvainGa/crystal-face/pull/8

Feel free to reject if I did it wrong, and thanks again for making that watch face. It's the main one I use.

SylvainGa committed:
2.18.9 Adjusted the fonts of the Venu Sq2

SylvainGa committed:
Fixed DEDUG for DEBUG

SylvainGa committed:
Merge pull request #8 from AdrienChenu/bug-local-city-time

Enable longitude west of Chicago and east of India

SylvainGa committed:
V2.19.0 Adds the following

- Added Altitude as a Complication
- Fixed bug in longitude typechecking for second time (thanks AdrienChenu)

SylvainGa committed:
2.18.12 (in progress)

- Fixed crash when complicationId is unexpectedly null
- Added Battery as a complication (to launch the battery widget)

SylvainGa committed:
V2.18.12 (in progress)

Found the cause for the crash at line 862.

onPress caught exception PreviousOperationNotCompleteException

Added try/catch around exitto calls.

SylvainGa committed:
Update README.md

SylvainGa committed:
2.18.13

- Compiled with CIQ 7.1.1

- Added Forerunner 165 and 165m

- Updated the Tesla status when the car is asleep to match the returned data from Tessie

- Fixed a broken weather icon

SylvainGa committed:
V2.18.4 Added the Descent MK3 variants

SylvainGa committed:
2.19.0

- Compiled with CIQ 7.1.1
- Added Active calories as a data fiel and goal. This value is calculated and not provided by Garmin so it might be off to what Garmin is showing. On Complication enabled watches, since it shows only Active calories, the calculated rested calories are added to the Calories fields and goals. To get the previous active calories, select Active Calories field or goal.

SylvainGa committed:
Update CrystalView.mc

SylvainGa committed:
Small optimization of the Complication code

SylvainGa committed:
Bugfix in Tesla code and new handling of Tesla data since 'asleep' has been replaced with 'offline'

SylvainGa committed:
Fix for Tesla error 412

SylvainGa committed:
Bug fix in Tesla online/offline detection

SylvainGa committed:
Update strings.xml

SylvainGa committed:
Debug code for TeslaInfo

SylvainGa committed:
Fixed timezone from preventing goals to be displayed

SylvainGa committed:
Updated readme

SylvainGa committed:
Update README.md

SylvainGa committed:
Added Enduro3 and new Fenix8 watches.  Small code optimisation for Telsa's complications

SylvainGa committed:
Added Recovery Time as a MoveBar option

SylvainGa committed:
MoveBar Recovery time can be a Complication and launch its widget when pressed

SylvainGa committed:
MoveBar Recovery time can be a Complication and launch its widget when pressed

Fixed the centering of the text when the colon between the hours and minutes is showing.

SylvainGa committed:
Doc update

SylvainGa committed:
Merge branch 'master' of https://github.com/SylvainGa/crystal-face

SylvainGa committed:
Removed weather station name

SylvainGa committed:
Added back OWM and the queueing of requests. Fixed for -403 error.

SylvainGa committed:
Previous code was working on the simulator but not a real device :-<

SylvainGa committed:
Show 200 for httpInternalErrorTesla when all is fine

SylvainGa committed:
No changes but compiled with CIQ 7.4.3

SylvainGa committed:
Update README.md

SylvainGa committed:
Duplicated 2.20.2 changes

SylvainGa committed:
Syntax error

SylvainGa committed:
Testing for empty data

SylvainGa committed:
Bug fix when Tesla data is a dict instead of a string. Enumerate data in onBackgroundData instead of looking only at the first one. Tesla background code not ran when reading from complication

SylvainGa committed:
Bug fix

SylvainGa committed:
Fail safe in case background data returns null

SylvainGa committed:
Fixed the MoveBar to not stay hidden when coming back from AOD

SylvainGa committed:
Update manifest.xml.prd

SylvainGa committed:
Update README.md

flowstatedev committed:
Support long press of live HR field to open HR glance

flowstatedev committed:
Fix build by removing BOM from resource xml files

SylvainGa committed:
Merge pull request #10 from flowstatedev/live-hr-onpress

Support long press of live HR field to open HR glance

SylvainGa committed:
Added new devices Approach50, Descent G2, Forerunner 570s/970 and VivoActive 6

SylvainGa committed:
Update README.md

SylvainGa committed:
Redid changes made in

SylvainGa committed:
2.22.0 - At request, there is now an option to have the date format to have a leading zero when there is only one digit.

SylvainGa committed:
Added Venu X1 device

SylvainGa committed:
Enable custon complication to be called from the battery data field and indicator

SylvainGa committed:
Added the new batterie complication setting string to all strings files
